### PR TITLE
Update dfe-analytics and switch on an entity table check

### PR DESCRIPTION
**Pull request by martyn-w**
(Original PR: https://github.com/DFE-Digital/schools-experience/pull/3061)

### Trello card
https://trello.com/c/dRaTmD3g/

### Context
Enable an [Entity Table Check job](https://github.com/DFE-Digital/dfe-analytics/blob/main/lib/dfe/analytics/entity_table_check_job.rb) on GIT which sends a row_count, 
a checksum and a timestamp for each entity table sent to BigQuery
This helps Data Insights team to confirm that database tables match BigQuery
and helps them debug them where they don’t

### Changes proposed in this pull request
- Update dfe-analytics gem to v1.12.5
- Enable entity table checks in DfE Analytics configuration

### Guidance to review
Check that:
- The gem is updated 
- enabled the check
- Things working as expected
- Scheduled the daily job

### CODE REVIEW

**Comment by martyn-w** [comment]
NB: updating the dfe-analytics gem to v1.12.7 throws a strange error in the tests:

```
An error occurred while loading spec_helper.
Failure/Error: require "dfe/analytics/testing"

NameError:
  uninitialized constant DfE::Analytics::BigQueryLegacyApi

      DfE::Analytics::BigQueryLegacyApi.singleton_class.send :prepend, LegacyTestOverrides
```

...whereas version v1.12.5 does not seem to have this problem

**Comment by asatwal** [comment]
@martyn-w 

Regarding the error: `uninitialized constant DfE::Analytics::BigQueryLegacyApi`

I believe  in [spec/rails_helper.rb](https://github.com/DFE-Digital/schools-experience/blob/master/spec/spec_helper.rb#L5) the line below maybe a problem:

require "dfe/analytics/testing"

I think it's because that file is referencing `DfE::Analytics::BigQueryLegacyApi`, whereas previously it didn't and the Gem is not fully loaded.

**Comment by martyn-w** [comment]
Thanks @asatwal, I've added two more require statements to [spec/spec_helper.rb](https://github.com/DFE-Digital/schools-experience/pull/3222/files#diff-89eebfcbc0f14b6d989517837ca1e94fce4e2ce9a03233641cd936f2b8d2ed94), have updated to v1.12.7 and the tests seem run fine now

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -2,6 +2,8 @@
 Dotenv.load '.env.test.local', '.env.test', '.env'
 
 require 'rspec/json_expectations'
+require "dfe/analytics/big_query_api"
+require "dfe/analytics/big_query_legacy_api"
```
Comment (asatwal) [inline spec/spec_helper.rb:6]: @ericaporter - Maybe we should be adding these lines to `dfe/analytics/testing.rb`

**Review by asatwal — APPROVED** [review]
👍 - LGTM

**Pull request merged**